code/datums/datumvars.dm

```diff
@@ -592,6 +593,30 @@ body
 			if(usr.client)
 				usr.client.cmd_assume_direct_control(M)
 
+		else if(href_list["offer_control"])
+			if(!check_rights(0))	return
+
+			var/mob/M = locate(href_list["offer_control"])
+			if(!istype(M))
+				usr << "This can only be used on instances of type /mob"
+				return
+			M << "Control of your mob has been offered to dead players."
+			log_admin("[key_name(usr)] has offered control of [M.real_name] to ghosts.")
+			message_admins("[key_name(usr)] has offered control of [M.real_name] to ghosts")

```

Comment:
key_name(usr) should be key_name_admin(usr) wherever it's seen by admins and M.real_name should be key_name_admin(M).